src/dataflow/components/stats.tsx

```diff
@@ -0,0 +1,276 @@
+          publishedDocOriginIds.push(pulishedDocEntry.originDoc);
+        }
+      });
+    }
+
+    // request the users
+    const usersPath = db.firebase.getUsersPath(db.stores.user);
+    const usersRef = db.firebase.ref(usersPath);
+    const usersSnapshot = await usersRef.once("value");
+    if (usersSnapshot) {
+      const users = usersSnapshot.val();
+      const userStats: StatsGroup[] = [];
+
+      Object.keys(users).forEach((userId) => {
+        const user = this.stores.class.getUserById(userId);
+        if (user && user.type === "student") {
+          const statsGroup: StatsGroup = {
+            studentName: user.fullName,
+            documents: 0,
+            programDocuments: 0,
+            runProgramDocuments: 0,
+            runProgramDataDocuments: 0,
+            runProgramRelayOnlyDocuments: 0,
+            publishedProgramDocuments: 0
+          };
+
+          const userEntry = users[userId];
+          Object.keys(userEntry.personalDocs).forEach((pDoc) => {
+            const document = userEntry.personalDocs[pDoc];
+
+            // determine if this document is a published document
+            if (publishedDocOriginIds.find(originDoc => originDoc === document.self.documentKey)) {
+              statsGroup.publishedProgramDocuments++;
+            }
+
+            // determine if this is a run program
+            if (document.properties && document.properties.dfRunId) {
+              statsGroup.runProgramDocuments++;
+              document.properties.dfHasData
+                ? statsGroup.runProgramDataDocuments++
+                : statsGroup.runProgramRelayOnlyDocuments++;
+            } else {
+              statsGroup.programDocuments++;
+            }
+            statsGroup.documents++;
+          });
+          userStats.push(statsGroup);
+        }
+      });
+      this.setState({ loading: false, userStats });
+    }
+  }
+
+  public render() {
+    return (
+      <div className="stats">
+        <div className="stats-title">
+        Classroom Statistics
+        </div>
+        { this.state.loading
+          ? <div className="loading">Loading classroom statistics...</div>
+          : this.renderStats()
+        }
+      </div>
+    );
+  }
+
+  private renderStats() {
+    return (
+      <div className="stats-container">
+        { this.renderTotalStats() }
+        { this.renderAverageStats() }
+        { this.renderUserStats() }
+        <button className="stats-button export" onClick={this.handleExport}>Export Student Stats (csv)</button>
+      </div>
+    );
+  }
+
+  private getStatTotals() {
+    const totals: StatsGroup = {
+      studentName: "",
+      documents: 0,
+      programDocuments: 0,
+      runProgramDocuments: 0,
+      runProgramDataDocuments: 0,
+      runProgramRelayOnlyDocuments: 0,
+      publishedProgramDocuments: 0
+    };
+    this.state.userStats.forEach(stats => {
+      totals.documents = totals.documents ? totals.documents + stats.documents : stats.documents;
```

Comment:
```suggestion
      totals.documents += stats.documents;
```
You've initialized the sums to zero, so you don't need the ternary here or in any of the subsequent lines.

Reply:
I had the statGroup properties as optional originally to avoid verbose instantiation, but I hated that route.  This was a remnant of that.  Thanks for catching.  Fixed.